src/main/java/roomescape/application/service/component/validator/CreateReservationValidator.java

```diff
@@ -0,0 +1,34 @@
+package roomescape.application.service.component.validator;
+
+import org.springframework.stereotype.Component;
+import roomescape.application.error.exception.CreateReservationValidateException;
+import roomescape.domain.reservation.vo.ReservationDate;
+import roomescape.domain.reservationtime.ReservationTime;
+import roomescape.repository.ReservationRepository;
+
+import static roomescape.application.error.code.ApplicationErrorCode.CANNOT_CREATE_EXIST_RESERVATION_AT_THIS_TIME;
+
+@Component
+public class CreateReservationValidator {
+
+    private static final String DATE_FORMAT = "yyyy-MM-dd";
+
+    private final ReservationRepository reservationRepository;
+
+    public CreateReservationValidator(ReservationRepository reservationRepository) {
+        this.reservationRepository = reservationRepository;
+    }
+
+    public void validate(ReservationDate reservationDate, ReservationTime reservationTime) {
+        reservationRepository.findByDateAndTimeId(reservationDate.date(), reservationTime.getId())
+                .ifPresent(entity -> {
+                    throw new CreateReservationValidateException(
+                            CANNOT_CREATE_EXIST_RESERVATION_AT_THIS_TIME,
```

Comment:
이건 단순 질문인데요.
ErrorCode를 Exception에 포함시키지 않고 enum으로 분리하면 어떤 장점이 있나요?

`CreateReservationValidateException` 의 경우 `CANNOT_CREATE_EXIST_RESERVATION_AT_THIS_TIME` 이 아닌 다른 ErrorCode를 갖게 되는 경우가 있나요?
혹은 이 예외에서 갖게되는 ErrorCode가 매우 다양한가요?

Reply:
CreateReservationValidateException과 같이 validate해야할 것이 늘어 날 수 있다고 생각했습니다.

위와 같이 존재하는 시간의 예약을 못생성하는 것과 같이
예약 생성에 대해 추가적으로 더 조건이 생길 수도 있습니다. 
이런 추가되는 조건 마다 Exception을 만들기보다는 enum으로 구분하는게 더 맞다고 생각했습니다.

Reply:
에러코드를 도입하신 이유에 대해선 공감이 가는데요.
지금의 구조에서는 다른 개발자가 Exception과 ErrorCode에 대해서 잘 알고 사용해야하고, 실수가 발생할 수 있어 보여요.
이 프로젝트에 익숙하지 않은 사람은 '난 예외를 던지고 싶은데 많은 ErrorCode 중 어떤 값을 사용해야 하고 메시지는 무엇으로 해야하는거지?' 에 대한 고민이 들 것 같아요.

아래와 같은 팩터리 메서드를 제공하면 이런 어려움이나 오류를 줄일 수 있을 것 같은데 어떠신가요?
메서드 명은 코파일럿의 도움을 받은거라ㅋㅋㅋ 구조 참고만 해주세요.

```java
public class CreateReservationTimeValidateException extends ApplicationException {
    private CreateReservationTimeValidateException(ApplicationErrorCode code) {
        super(code);
    }

    private CreateReservationTimeValidateException(ApplicationErrorCode code, String message) {
        super(code, message);
    }

    public static CreateReservationTimeValidateException cannotCreateExistReservationAtThisTime() {
        return new CreateReservationTimeValidateException(ApplicationErrorCode.CANNOT_CREATE_EXIST_RESERVATION_AT_THIS_TIME);
    }
}
```

```suggestion
                            throw CreateReservationTimeValidateException.cannotCreateExistReservationAtThisTime();
```

(+) 예외 메시지에 대해서도, 예외를 사용하는 개발자가 메시지 내용을 다 지정해야하는 것일지 같이 고민해보셨으면 해요~

Reply:
에러 코드를 외부애서 직접 넣어주기보다 내부에서 정적 메소드로 호출하는 방식을 활용할 수 있겠네요!
좋은 인사이트 감사합니다. 가독성 측면에 있어서도 주신 의견이 더 좋은 것 같아요!

Reply:
메세지를 넘겼던 이유는 메세지의 Key 값을 남겨 
예외에 대한 메세지를 로그에 남기거나 할때 추적에 용이하게 하기위해서 넣었는데, 생각해보니 Key값만 넘기도록 하는 게 더 좋을 것 같네요!

Reply:
예외 메시지 로깅 중요하죠!
메시지 포맷은 대개 달라지지 않을거라 말씀 주신 것처럼 key 만 넘기는 것이 좋습니다 ㅎㅎ